### ISSUE TITLE
Function App Does Not Respond on HTTP Trigger

### ISSUE THREAD

**Issue by marcofortin** [Azure/azure-functions-host#5810]
I have a Function App setup in JavaScript that gets triggered by an HTTP request from a MS Power Automate/Flow. The script runs perfectly when running from the Azure portal or making an HTTP request from Postman. However, the Function App seems to go to 'sleep' when it has not been triggered in a while. I am getting this error by MS Power Automate/Flow: 
`BadRequest. Http request failed: the server did not respond within the timeout limit. Please see logic app limits at https://aka.ms/logic-apps-limits-and-config#http-limits.`
But, the script normally runs in less than a minute so it must be that it did not run. Is there something I can do to prevent the 'sleeping' from the script and guarantee a response at every request? The HTTP requests are automated daily and I have tried making another HTTP request prior to all other request to hopefully 'wake-up' the script but that did not work. 

Thanks!

**Comment by gzuber** [comment]
@marcofortin Can I get some more details about your function app (name, region, etc...) so I can look into this from our end? I might be able to see what's causing the timeout and we can take it from there. Thanks!

**Comment by marcofortin** [comment]
Hey @gzuber !
Name: agony-bot
Region: Canada Central
Subscription Id: 949ab995-2999-4bcf-9bd6-a9de50440692
Ressource group: innovation
Thanks!!!

**Comment by marcofortin** [comment]
@gzuber Are there any more details you need to look into this problem?

**Comment by marcofortin** [comment]
@gzuber Any news?

**Comment by gzuber** [comment]
@marcofortin I'm so sorry it took my so long to get back to you -- are you still seeing this issue? I've looked at the logs on our end and I saw some timeouts, but I also saw this error:

```
[error] Worker 9f4c8746-9961-4a5a-8a42-7dba66ce8e0d uncaught exception (learn more: https://go.microsoft.com/fwlink/?linkid=2097909 ): ReferenceError: reject is not defined
     at Request.userCallback (/home/site/wwwroot/AgonyCalculator/tools.js:30:17)
```

Have you seen that exception? I'm not totally convinced it's related to the timeouts, but it's something that we would want to fix before diving deeper just in case since it's happening in conjunction with timeouts.
